### PR TITLE
fix null pointer exception

### LINKED ISSUE #264: NullPointerException while scanning personal projects
### Jenkins and plugins versions report

<details>
  <summary>Environment</summary>

  ```text
Jenkins: 2.375.1
OS: Linux - 5.13.19-2-MANJARO
---
ace-editor:1.1
ant:481.v7b_09e538fcca
antisamy-markup-formatter:155.v795fb_8702324
apache-httpcomponents-client-4-api:4.5.13-138.v4e7d9a_7b_a_e61
bootstrap5-api:5.2.1-3
bouncycastle-api:2.26
branch-api:2.1051.v9985666b_f6cc
build-timeout:1.28
caffeine-api:2.9.3-65.v6a_47d0f4d1fe
checks-api:1.8.1
cloudbees-folder:6.800.v71307ca_b_986b
command-launcher:90.v669d7ccb_7c31
commons-lang3-api:3.12.0-36.vd97de6465d5b_
commons-text-api:1.10.0-27.vb_fa_3896786a_7
credentials:1214.v1de940103927
credentials-binding:523.vd859a_4b_122e6
display-url-api:2.3.7
durable-task:503.v57154d18d478
echarts-api:5.4.0-1
email-ext:2.93
font-awesome-api:6.2.1-1
git:5.0.0
git-client:4.0.0
github:1.36.1
github-api:1.303-400.v35c2d8258028
github-branch-source:1696.v3a_7603564d04
gitlab-api:5.0.1-78.v47a_45b_9f78b_7
gitlab-branch-source:644.va_a_66886e07b_5
gradle:2.2
handy-uri-templates-2-api:2.1.8-22.v77d5b_75e6953
instance-identity:142.v04572ca_5b_265
ionicons-api:31.v4757b_6987003
jackson2-api:2.14.1-313.v504cdd45c18b
jakarta-activation-api:2.0.1-2
jakarta-mail-api:2.0.1-2
javax-activation-api:1.2.0-5
javax-mail-api:1.6.2-8
jaxb:2.3.7-1
jdk-tool:63.v62d2fd4b_4793
jersey2-api:2.38-1
jjwt-api:0.11.5-77.v646c772fddb_0
jquery3-api:3.6.1-2
jsch:0.1.55.61.va_e9ee26616e7
junit:1166.va_436e268e972
ldap:2.12
mailer:438.v02c7f0a_12fa_4
matrix-auth:3.1.6
matrix-project:785.v06b_7f47b_c631
mina-sshd-api-common:2.9.2-50.va_0e1f42659a_a
mina-sshd-api-core:2.9.2-50.va_0e1f42659a_a
momentjs:1.1.1
okhttp-api:4.9.3-108.v0feda04578cf
pam-auth:1.10
pipeline-build-step:2.18
pipeline-github-lib:38.v445716ea_edda_
pipeline-graph-analysis:195.v5812d95a_a_2f9
pipeline-groovy-lib:629.vb_5627b_ee2104
pipeline-input-step:466.v6d0a_5df34f81
pipeline-milestone-step:101.vd572fef9d926
pipeline-model-api:2.2118.v31fd5b_9944b_5
pipeline-model-definition:2.2118.v31fd5b_9944b_5
pipeline-model-extensions:2.2118.v31fd5b_9944b_5
pipeline-rest-api:2.28
pipeline-stage-step:296.v5f6908f017a_5
pipeline-stage-tags-metadata:2.2118.v31fd5b_9944b_5
pipeline-stage-view:2.28
plain-credentials:143.v1b_df8b_d3b_e48
plugin-util-api:2.20.0
popper2-api:2.11.6-2
resource-disposer:0.20
scm-api:631.v9143df5b_e4a_a
script-security:1228.vd93135a_2fb_25
snakeyaml-api:1.33-90.v80dcb_3814d35
ssh-credentials:305.v8f4381501156
ssh-slaves:2.854.v7fd446b_337c9
sshd:3.275.v9e17c10f2571
structs:324.va_f5d6774f3a_d
timestamper:1.21
token-macro:321.vd7cc1f2a_52c8
trilead-api:2.84.v72119de229b_7
variant:59.vf075fe829ccb
workflow-aggregator:590.v6a_d052e5a_a_b_5
workflow-api:1200.v8005c684b_a_c6
workflow-basic-steps:994.vd57e3ca_46d24
workflow-cps:3601.v9b_36a_d99e1cc
workflow-durable-task-step:1217.v38306d8fa_b_5c
workflow-job:1254.v3f64639b_11dd
workflow-multibranch:716.vc692a_e52371b_
workflow-scm-step:400.v6b_89a_1317c9a_
workflow-step-api:639.v6eca_cd8c04a_a_
workflow-support:839.v35e2736cfd5c
ws-cleanup:0.44
  ```
</details>

### What Operating System are you using (both controller, and any agents involved in the problem)?

Linux OS

### Reproduction steps

Create a Personal Access Token with scope `api` (I am using gitlab.com but I think this is not relevant)
Create a new "Organization Folder"
Configure this project to scan your personal repositories (and not a group)

User URL https://gitlab.com/jmini

(since then I have

### Expected Results

The repositories are found

### Actual Results

NullPointerException while running "Scan GitLab Group Now"

```
[Fri Jan 06 13:51:03 CET 2023] Starting organization scan...
[Fri Jan 06 13:51:03 CET 2023] Updating actions...
User URL: [Jérémie Bresson](https://gitlab.com/jmini)
[Fri Jan 06 13:51:03 CET 2023] Consulting GitLab Group

Checking projects...
ERROR: [Fri Jan 06 13:51:04 CET 2023] Could not fetch sources from navigator io.jenkins.plugins.gitlabbranchsource.GitLabSCMNavigator@1d9cb17f
[Fri Jan 06 13:51:04 CET 2023] Finished organization scan. Scan took 0.76 sec
FATAL: Failed to recompute children of Jérémie Bresson
java.lang.NullPointerException
	at io.jenkins.plugins.gitlabbranchsource.GitLabSCMNavigator.visitSources(GitLabSCMNavigator.java:271)
	at jenkins.branch.OrganizationFolder.computeChildren(OrganizationFolder.java:536)
	at com.cloudbees.hudson.plugins.folder.computed.ComputedFolder.updateChildren(ComputedFolder.java:278)
	at com.cloudbees.hudson.plugins.folder.computed.FolderComputation.run(FolderComputation.java:166)
	at jenkins.branch.OrganizationFolder$OrganizationScan.run(OrganizationFolder.java:918)
	at hudson.model.ResourceController.execute(ResourceController.java:107)
	at hudson.model.Executor.run(Executor.java:449)
Finished: FAILURE
```

The NPE is because of this line:
https://github.com/jenkinsci/gitlab-branch-source-plugin/blob/aa66886e07b57d7b350c262114a68f0608712f25/src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabSCMNavigator.java#L271

`p.getArchived()` returns a Boolean and is obviously set to `null`

### Anything else?

I am not sure how this can happen, because when I executed the same query against Gitlab using this [jbang](https://www.jbang.dev/) script:

<details>
  <summary>script to call gitLabApi.getProjectApi().getUserProjects(..)</summary>

```java
///usr/bin/env jbang "$0" "$@" ; exit $?

//JAVA 17
//DEPS org.gitlab4j:gitlab4j-api:5.0.1

import java.util.List;

import org.gitlab4j.api.GitLabApi;
import org.gitlab4j.api.models.Project;
import org.gitlab4j.api.models.ProjectFilter;

public class GitLabTest {
	private static final String URL = "https://gitlab.com";
	private static final String TOKEN = "glpat-********************";

	public static void main(String... args) throws Exception {
		GitLabApi gitLabApi = new GitLabApi(URL, TOKEN, null, null);
		List<Project> projects = gitLabApi.getProjectApi().getUserProjects("jmini", new ProjectFilter().withOwned(true));
		System.out.println(projects);
		for (Project project : projects) {
			if (project.getArchived() == null) {
				System.out.println("Project with error:" + project);
			}
		}
	}
}
```
</details>

All my projects have a value for `project.getArchived()`. So I am not sure how this NPE is possible.

---

My work-around is to use a group:
https://gitlab.com/jmini-test-group

Instead of trying to import personal repositories.

### PR DESCRIPTION
**Pull request by rhit-gawronja**
fixes #264 

### Testing done
Completed reproduction steps before and after, passed automatic testing coverage. Where these fixes differ is when you have an archived project  previously it would cause a NullPointerException in the scanning process. This issue has been remediated by skipping over null projects in the scanning. Issue link: https://github.com/jenkinsci/gitlab-branch-source-plugin/issues/264

```[tasklist]
### Submitter checklist
- [ x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ x] Ensure that the pull request title represents the desired changelog entry
- [ x] Please describe what you did
- [ x] Link to relevant issues in GitHub or Jira
- [ x] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

File: src/test/java/io/jenkins/plugins/gitlabbranchsource/GitLabHookCreatorParameterizedTest.java [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package io.jenkins.plugins.gitlabbranchsource;
+ackage io.jenkins.plugins.gitlabbranchsource;
```
Comment (UltimateGeek) [inline src/test/java/io/jenkins/plugins/gitlabbranchsource/GitLabHookCreatorParameterizedTest.java:1]: Need to revert this change

```suggestion
package io.jenkins.plugins.gitlabbranchsource;
```

File: src/test/java/io/jenkins/plugins/gitlabbranchsource/.GitLabHookCreatorParameterizedTest.java.~undo-tree~ [hunk 2]
```diff
@@ -0,0 +1,5 @@
+(undo-tree-save-format-version . 1)
+"7b5284b701661ee0b2dde7edf27712548c6ad1b0"
+[nil nil nil nil (26009 29458 405571 664000) 0 nil]
+([nil current (("p" . 1) (t 26009 23814 550346 484000)) nil (26009 29458 405567 967000) 0 nil])
+nil
```
Comment (UltimateGeek) [inline src/test/java/io/jenkins/plugins/gitlabbranchsource/.GitLabHookCreatorParameterizedTest.java.~undo-tree~:5]: remove this file

File: src/main/java/io/jenkins/plugins/gitlabbranchsource/.GitLabSCMNavigator.java.~undo-tree~
Comment (UltimateGeek) [inline src/main/java/io/jenkins/plugins/gitlabbranchsource/.GitLabSCMNavigator.java.~undo-tree~:1]: remove this file

**Review by UltimateGeek — COMMENTED** [review]
I recommend not adding extra comments above the functions that don't provide any more information than the name of the function already provides.

File: src/test/java/io/jenkins/plugins/gitlabbranchsource/GitLabHookCreatorParameterizedTest.java (hunk 1 above)
Comment (rhit-gawronja) [reply src/test/java/io/jenkins/plugins/gitlabbranchsource/GitLabHookCreatorParameterizedTest.java]: Done, thanks for the feedback

File: src/test/java/io/jenkins/plugins/gitlabbranchsource/.GitLabHookCreatorParameterizedTest.java.~undo-tree~ (hunk 2 above)
Comment (rhit-gawronja) [reply src/test/java/io/jenkins/plugins/gitlabbranchsource/.GitLabHookCreatorParameterizedTest.java.~undo-tree~]: Done, thanks for the feedback

File: src/main/java/io/jenkins/plugins/gitlabbranchsource/.GitLabSCMNavigator.java.~undo-tree~
Comment (rhit-gawronja) [reply src/main/java/io/jenkins/plugins/gitlabbranchsource/.GitLabSCMNavigator.java.~undo-tree~:1]: Done, thanks for the feedback

**Comment by rhit-gawronja** [comment]
#264   Small fix for this issue

File: src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabSCMNavigator.java [hunk 3]
```diff
@@ -260,8 +262,14 @@ public void visitSources(@NonNull final SCMSourceObserver observer) throws IOExc
...
                 webHookUrl = GitLabHookCreator.getHookUrl(server, true);
             }
+            // we may check with web hooks but we don't check if the projects returned are null
+
             for (Project p : projects) {
                 count++;
+                if (p == null) {
+                    System.out.println("null project");
+                    continue;
+                }
```
Comment (jetersen) [inline src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabSCMNavigator.java:272]: Loose the println and simply filter them.

```suggestion
            projects = projects.stream().filter(Objects::nonNull).collect(Collectors.toList());
            for (Project p : projects) {
                count++;
```

File: src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabSCMNavigator.java [hunk 4]
```diff
@@ -194,6 +196,7 @@ public void setTraits(@CheckForNull SCMTrait[] traits) {
         }
     }
 
+    // gets the owner of the gitlab
```
Comment (jetersen) [inline src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabSCMNavigator.java:199]:
```suggestion
```

File: src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabSCMNavigator.java [hunk 5]
```diff
@@ -229,6 +232,7 @@ protected String id() {
         return getServerUrlFromName(serverName) + "::" + projectOwner;
     }
 
+    // this checks all of our project source files
```
Comment (jetersen) [inline src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabSCMNavigator.java:235]:
```suggestion
```

File: src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabSCMNavigator.java [hunk 6]
```diff
@@ -260,8 +264,12 @@ public void visitSources(@NonNull final SCMSourceObserver observer) throws IOExc
                         serverUrl, getPrivateTokenAsPlainText(webHookCredentials), null, getProxyConfig(serverUrl));
                 webHookUrl = GitLabHookCreator.getHookUrl(server, true);
             }
+            // we may check with web hooks but we don't check if the projects returned are null
```
Comment (jetersen) [inline src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabSCMNavigator.java:267]:
```suggestion
```

File: src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabSCMNavigator.java [hunk 7]
```diff
@@ -260,8 +264,12 @@ public void visitSources(@NonNull final SCMSourceObserver observer) throws IOExc
...
+            projects = projects.stream().filter(Objects::nonNull).collect(Collectors.toList());
+
```
Comment (jetersen) [inline src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabSCMNavigator.java:269]:
```suggestion
```

File: src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabSCMNavigator.java [hunk 8]
```diff
@@ -260,8 +264,12 @@ public void visitSources(@NonNull final SCMSourceObserver observer) throws IOExc
...
             for (Project p : projects) {
                 count++;
+
```
Comment (jetersen) [inline src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabSCMNavigator.java:272]:
```suggestion
```

**Review by jetersen — CHANGES_REQUESTED** [review]

**Review by jetersen — APPROVED** [review]

**Pull request merged**